### ISSUE TITLE
Since KSP 1.1.1 update, references to stage resources fail with NRE

### ISSUE THREAD

**Issue by asazernik** [KSP-KOS/KOS#1625]
I re-ran a script that worked in 1.1.0 to verify that this is a new bug. This only applies to the resource-related suffixes, not to other suffixes of the stage object.

A quick console log to explain better:

```
print stage:number.
3
print stage:resources.
Object reference not set to an instance of an object
At interpreter history, line 3
print stage:resources
            ^
print stage:resourceslex.
Object reference not set to an instance of an object
At interpreter history, line 4
print stage:resourceslex
            ^
```

**Comment by hvacengi** [comment]
Could you please post your output_log.txt file as well?  (via dropbox, pastebin, or other such)

**Comment by asazernik** [comment]
Was just about to do that :-) The relevant section is below; I'll post the link as soon as I get it uploaded.

```
[LOG 18:18:34.252] kOS: At interpreter history, line 4
print stage:resourceslex.
            ^

[LOG 18:18:34.253] System.NullReferenceException: Object reference not set to an instance of an object
  at Vessel.GetActiveResources () [0x00000] in <filename unknown>:0 
  at kOS.Suffixed.StageValues.GetResourceDictionary () [0x00000] in <filename unknown>:0 
  at kOS.Safe.Encapsulation.Suffixes.Suffix`1[kOS.Safe.Encapsulation.Lexicon].Get () [0x00000] in <filename unknown>:0 
  at kOS.Safe.Encapsulation.Structure.GetSuffix (System.String suffixName) [0x00000] in <filename unknown>:0 
  at kOS.Suffixed.StageValues.GetSuffix (System.String suffixName) [0x00000] in <filename unknown>:0 
  at kOS.Safe.Compilation.OpcodeGetMember.Execute (ICpu cpu) [0x00000] in <filename unknown>:0 
  at kOS.Safe.Execution.CPU.ExecuteInstruction (IProgramContext context, Boolean doProfiling) [0x00000] in <filename unknown>:0 
[LOG 18:18:34.253] Code Fragment
File                 Line:Col IP   label   opcode operand
====                 ====:=== ==== ================================  
interpreter history     1:13  0006 @0006   pop 
interpreter history     2:1   0007 @0007   push _KOSArgMarker_ 
interpreter history     2:7   0008 @0008   push $stage 
interpreter history     2:13  0009 @0009   push number 
interpreter history     2:13  0010 @0010   getmember 
interpreter history     2:13  0011 @0011   call print() 
interpreter history     2:13  0012 @0012   pop 
interpreter history     3:1   0013 @0013   push _KOSArgMarker_ 
interpreter history     3:7   0014 @0014   push $stage 
interpreter history     3:13  0015 @0015   push resources 
interpreter history     3:13  0016 @0016   getmember 
interpreter history     3:13  0017 @0017   call print() 
interpreter history     3:13  0018 @0018   pop 
interpreter history     4:1   0019 @0019   push _KOSArgMarker_ 
interpreter history     4:7   0020 @0020   push $stage 
interpreter history     4:13  0021 @0021   push resourceslex 
interpreter history     4:13  0022 @0022   getmember <<--INSTRUCTION POINTER--
interpreter history     4:13  0023 @0023   call print() 
interpreter history     4:13  0024 @0024   pop 
                        0:0   0025         EOF 

[LOG 18:18:34.253] kOS: Stack dump: stackPointer = 0
000 SP-> _KOSArgMarker_ (type: KOSArgMarkerType)
```

**Comment by asazernik** [comment]
[Here you go](https://gist.github.com/asazernik/062537716fffff3d426f4f431bce752c).

**Comment by hvacengi** [comment]
I just put together that you identified the new version of KSP 1.1.1 (which I honestly didn't know was released until about 30 minutes ago).  I've been testing most of the features through the experimentals channel, but I didn't see this break.  We'll have to look and see if it's a simple re-compile or if it needs something more complicated.

**Comment by asazernik** [comment]
Heh, yeah, I saw the update queued in steam earlier today, and kicked it off early (it's been that kind of day).

**Comment by hvacengi** [comment]
It looks like this has been fixed by squad themselves in KSP 1.1.2.  Can you confirm?

**Comment by asazernik** [comment]
All fixed!

On Mon, May 2, 2016 at 4:55 AM Brad White <notifications@github.com> wrote:

> It looks like this has been fixed by squad themselves in KSP 1.1.2. Can
> you confirm?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/KSP-KOS/KOS/issues/1625#issuecomment-216217913>
>
-- 

Sent from my phone. Expect few words and many typos.
